### ISSUE TITLE
Bug for argument `use_decision_function`

### ISSUE THREAD

**Issue by Make42** [globality-corp/sklearn-hierarchical-classification#45]
I think I found another bug: For Classifiers like LinearSVC, the methods like [`decision_function`](https://scikit-learn.org/stable/modules/generated/sklearn.svm.LinearSVC.html#sklearn.svm.LinearSVC.decision_function) returns only **one** value, if there are only two classes. Thus the [code](https://github.com/globality-corp/sklearn-hierarchical-classification/blob/1de19f782d992a82dace895f9c24a0fc074baeeb/sklearn_hierarchical_classification/classifier.py#L583)

            if self.use_decision_function and hasattr(clf, "decision_function"):
                if self.feature_extraction == "raw":
                    probs = clf.decision_function([x])
                    argmax = np.argmax(probs)
                    score = probs[0, argmax]

                else:
                    probs = clf.decision_function(x)
                    argmax = np.argmax(probs)
                    score = probs[argmax]

should actually be

            if self.use_decision_function and hasattr(clf, "decision_function"):
                probs = clf.decision_function([x]) if self.feature_extraction == "raw" else clf.decision_function(x)
                last_axis = probs.ndim-1
                if probs.ndim == 1 and probs.shape[0] == 1:
                    probs = np.concatenate((-probs, probs), axis=last_axis)
                argmax = probs.argmax(axis=last_axis)
                score = probs[argmax] if last_axis == 0 else probs[0, argmax]

**Comment by Make42** [comment]
This is the same bug as https://github.com/globality-corp/sklearn-hierarchical-classification/issues/44 - I missed this at first.
